client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -12,7 +12,60 @@ import { translate } from 'i18n-calypso';
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 import { errorNotice } from 'state/notices/actions';
 import { http } from 'state/data-layer/wpcom-http/actions';
-import { JETPACK_ONBOARDING_SETTINGS_SAVE } from 'state/action-types';
+import {
+	JETPACK_ONBOARDING_SETTINGS_REQUEST,
+	JETPACK_ONBOARDING_SETTINGS_SAVE,
+} from 'state/action-types';
+import { receiveJetpackOnboardingSettings as receiveJetpackOnboardingSettingsAction } from 'state/jetpack-onboarding/actions';
+
+export const announceRequestFailure = ( { dispatch } ) =>
+	dispatch(
+		errorNotice( translate( 'Could not fetch settings from site. Please try again later.' ) )
+	);
+
+const receiveJetpackOnboardingSettings = ( { dispatch }, action, { data: settings } ) => {
+	const { siteId } = action;
+
+	if ( ! settings ) {
+		return announceRequestFailure( { dispatch }, action );
+	}
+
+	dispatch( receiveJetpackOnboardingSettingsAction( siteId, settings ) );
+};
+
+/**
+ * Dispatches a request to fetch settings for a given site
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const requestJetpackOnboardingSettings = ( { dispatch, getState }, action ) => {
+	const { siteId } = action;
+	const state = getState();
+	const token = get( state.jetpackOnboarding.credentials, [ siteId, 'token' ], null );
+	const jpUser = get( state.jetpackOnboarding.credentials, [ siteId, 'userEmail' ], null );
```

Comment:
merp. :wink:

```js
const token = get( state, [ 'jetpackOnboarding', 'credentials', siteId, 'token' ] );
const jpUser = get( state, [ 'jetpackOnboarding', 'credentials', siteId, 'userToken' ] );
```

Reply:
Hehe, copypasta. I think we have a selector by now, I'll try to find it...

Reply:
1d83da9